$NetBSD: patch-aa,v 1.4 2013/07/03 08:19:53 wiz Exp $

fix a problem with the curses widget and non-UTF8 locales: keys which
used less bytes in the system encoding than in the UTF8 passed to
the caller caused that the passphrase got truncated because only
the input length was accounted for

--- pinentry/pinentry-curses.c.orig	2009-04-16 15:06:53.000000000 +0000
+++ pinentry/pinentry-curses.c
@@ -819,8 +819,10 @@ dialog_run (pinentry_t pinentry, const c
       if (pin_utf8)
 	{
 	  pinentry_setbufferlen (pinentry, strlen (pin_utf8) + 1);
-	  if (pinentry->pin)
+	  if (pinentry->pin) {
 	    strcpy (pinentry->pin, pin_utf8);
+	    diag.pin_len = strlen (pin_utf8);
+	  }
 	  secmem_free (pin_utf8);
 	  pinentry->locale_err = 0;
 	}
